### PR TITLE
feat: Add pointer events

### PR DESCRIPTION
**Pull request by eps1lon**
**What**:

Add [Pointer events](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events) to `fireEvent`.

**Why**:

Consistent with other implemented events

**How**:

Add each event in https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events#Event_types_and_Global_Event_Handlers and use defaults as described in the corresponding event page e.g. https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/gotpointercapture_event

**Checklist**:

- ~[ ] Documentation added to the
      [docs site](https://github.com/alexkrolick/testing-library-docs)~ Don't think that's necessary since I didn't find docs for specific events.
- [x] Typescript definitions updated
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

File: src/events.js [hunk 1]
```diff
@@ -297,6 +297,47 @@ const eventMap = {
...
+  // pointer events
+  pointerover: {
+    EventType: 'PointerEvent',
+    defaultInit: {bubbles: true, cancelable: true},
+  },
+  pointerenter: {
+    EventType: 'PointerEvent',
+    defaultInit: {bubbles: false, cancelable: false},
+  },
+  pointerdown: {
```
Comment (eps1lon) [inline src/events.js]: Not familiar why the decision was made to name it `keyDown` instead of `keydown` and if it applies here as well.

Comment (kentcdodds) [reply src/events.js]: I think we wanted to camelCase it similar to how react does 👍 I believe it applies here as well.

Comment (eps1lon) [reply src/events.js]: Alright will do.

Comment (eps1lon) [reply src/events.js]: Done

**Comment by kentcdodds** [comment]
@all-contributors please add @eps1lon for tests

**Comment by kentcdodds** [comment]
Thanks so much for your help! I've added you as a collaborator on the project. Please make sure that you review the `other/MAINTAINING.md` and `CONTRIBUTING.md` files (specifically the bit about the commit messages and the git hooks) and familiarize yourself with the code of conduct (we're using [the contributor covenant](http://contributor-covenant.org/)). You might also want to [watch the repo](https://help.github.com/articles/watching-repositories/) to be notified when someone files an issue/PR. Please continue to make PRs as you feel the need (you can make your branches directly on the repo rather than your fork if you want). Thanks! And welcome to the team :)

**Pull request merged**